model/qti/ParserFactory.php

```diff
@@ -899,7 +900,7 @@ protected function buildResponseDeclaration(DOMElement $data)
 
             $mapping = [];
             foreach ($mappingNode->mapEntry as $mapEntry) {
-                $mapping[(string) $mapEntry['mapKey']] = (string) $mapEntry['mappedValue'];
+                $mapping[(string) htmlspecialchars($mapEntry['mapKey'])] = (string) $mapEntry['mappedValue'];
```

Comment:
It would be nice to add a test for this, since it cause headache :) 

I know we have an integration test: taoQtiItem/test/integration/ParserFactoryTest.php

Reply:
Got it. My suggestion was just to extend the xml for the integration test with this mapKey and see if it passes. I did not ask to create new unit tests :)